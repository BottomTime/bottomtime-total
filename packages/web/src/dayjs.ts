import dayjs from 'dayjs';
import localizedFormat from 'dayjs/plugin/localizedFormat';
import relativeTime from 'dayjs/plugin/relativeTime';
import tz from 'dayjs/plugin/timezone';
import utc from 'dayjs/plugin/utc';

dayjs.extend(localizedFormat);
dayjs.extend(relativeTime);
dayjs.extend(tz);
dayjs.extend(utc);

export default dayjs;
