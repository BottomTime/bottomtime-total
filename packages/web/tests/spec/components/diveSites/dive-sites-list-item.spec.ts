import { ComponentMountingOptions, mount } from '@vue/test-utils';

import { Pinia, createPinia } from 'pinia';
import DiveSitesListItem from 'src/components/diveSites/dive-sites-list-item.vue';
import 'tests/dayjs';

import {
  DiveSiteWithFullProperties,
  DiveSiteWithMinimalProperties,
} from '../../../fixtures/sites';
import StarRatingStub from '../../../stubs/star-rating-stub.vue';

describe('Dive Sites List Item component', () => {
  let pinia: Pinia;
  let global: ComponentMountingOptions<typeof DiveSitesListItem>['global'];

  beforeEach(() => {
    pinia = createPinia();
    global = {
      plugins: [pinia],
      stubs: {
        StarRating: StarRatingStub,
      },
    };
  });

  it('will render a dive site with minimal properties', () => {
    const wrapper = mount(DiveSitesListItem, {
      props: { site: DiveSiteWithMinimalProperties },
      global,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('will render a dive site with full properties', () => {
    const wrapper = mount(DiveSitesListItem, {
      props: { site: DiveSiteWithFullProperties },
      global,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('will emit event when name of site is clicked', async () => {
    const wrapper = mount(DiveSitesListItem, {
      props: { site: DiveSiteWithMinimalProperties },
      global,
    });

    await wrapper
      .get(`[data-testid="select-site-${DiveSiteWithMinimalProperties.id}"]`)
      .trigger('click');

    expect(wrapper.emitted('site-selected')).toEqual([
      [DiveSiteWithMinimalProperties],
    ]);
  });
});
