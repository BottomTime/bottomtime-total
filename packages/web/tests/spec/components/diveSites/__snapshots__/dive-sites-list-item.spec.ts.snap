// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dive Sites List Item component will render a dive site with full properties 1`] = `
"<div class="flex flex-col gap-0 bg-gradient-to-b from-blue-200 to-blue-400 dark:from-blue-800 dark:to-blue-950 shadow-xl rounded-md border-blue-900 border-opacity-75 border-2 relative"><a class="absolute -top-32" name="e25add45-3bf3-4be5-9859-7ea516f63f00"></a><!-- Site Name -->
  <div class="text-center bg-blue-200 rounded-t-md"><button class="text-link hover:text-link-hover py-2 mt-1 mb-1 text-sm" data-testid="select-site-e25add45-3bf3-4be5-9859-7ea516f63f00" type="button">
      <!--v-if-->
      <h1 class="text-2xl font-bold">webbed, defiant polyester</h1>
    </button></div>
  <div class="flex flex-col gap-3 p-3 grow">
    <!-- Location and creator -->
    <div class="flex align-baseline items-baseline">
      <p class="grow"><span class="text-danger mr-2"><i class="fas fa-map-marker-alt fa-fw"></i></span><span>Waldorf, ME, JO</span></p>
      <p><button class="text-link hover:text-link-hover py-2 mt-1 mb-1 text-lg" data-testid="site-creator-e25add45-3bf3-4be5-9859-7ea516f63f00" type="button">
          <!--v-if--><span class="mr-2"><i class="fas fa-user fa-fw"></i></span><span>@Constance_Walker19</span>
        </button></p>
    </div><!-- Description -->
    <div class="text-sm text-justify italic">Possimus animi voluptate repellendus hic facilis dolorem. Animi natus perspiciatis ut voluptate velit eos vel nihil doloremque. Dolorem tenetur magnam maxime. Aut doloribus eveniet corporis.</div><!-- Shim to force the rest of the content to the bottom -->
    <div class="grow"></div><!-- Depth, shore access, free to dive -->
    <div class="flex gap-1.5 justify-evenly">
      <div class="text-center">
        <p class="font-bold">Depth:</p>
        <p class="text-sm italic"><span>21.62 m</span></p>
      </div>
      <div class="text-center">
        <p class="font-bold">Shore Access:</p>
        <p class="text-sm italic">Yes</p>
      </div>
      <div class="text-center">
        <p class="font-bold">Free to Dive:</p>
        <p class="text-sm italic">Yes</p>
      </div>
    </div><!-- Rating and difficulty -->
    <div class="flex items-center justify-evenly">
      <div class="text-center">
        <p class="font-bold">Rating:</p>
        <p>
        <div class="flex gap-1.5 text-sm place-items-baseline">
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div><span>5.00</span>
        </div>
        </p>
      </div>
      <div class="grow"></div>
      <div class="text-center">
        <p class="font-bold">Difficulty:</p>
        <p>
        <div class="flex gap-1.5 text-sm place-items-baseline">
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="text-warn-hover fa-layers"><i class="fas fa-star"></i></span></div>
          <div><span class="fa-layers text-grey-500"><i class="far fa-star"></i></span></div><span>4.40</span>
        </div>
        </p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Dive Sites List Item component will render a dive site with minimal properties 1`] = `
"<div class="flex flex-col gap-0 bg-gradient-to-b from-blue-200 to-blue-400 dark:from-blue-800 dark:to-blue-950 shadow-xl rounded-md border-blue-900 border-opacity-75 border-2 relative"><a class="absolute -top-32" name="90b06d4a-605c-49d0-8738-cf557cf34b0a"></a><!-- Site Name -->
  <div class="text-center bg-blue-200 rounded-t-md"><button class="text-link hover:text-link-hover py-2 mt-1 mb-1 text-sm" data-testid="select-site-90b06d4a-605c-49d0-8738-cf557cf34b0a" type="button">
      <!--v-if-->
      <h1 class="text-2xl font-bold">remorseful, spanish demon</h1>
    </button></div>
  <div class="flex flex-col gap-3 p-3 grow">
    <!-- Location and creator -->
    <div class="flex align-baseline items-baseline">
      <p class="grow"><span class="text-danger mr-2"><i class="fas fa-map-marker-alt fa-fw"></i></span><span>Hudsonstad, NV, HM</span></p>
      <p><button class="text-link hover:text-link-hover py-2 mt-1 mb-1 text-lg" data-testid="site-creator-90b06d4a-605c-49d0-8738-cf557cf34b0a" type="button">
          <!--v-if--><span class="mr-2"><i class="fas fa-user fa-fw"></i></span><span>@Grayson.Wolff58</span>
        </button></p>
    </div><!-- Description -->
    <div class="text-sm text-justify italic"></div><!-- Shim to force the rest of the content to the bottom -->
    <div class="grow"></div><!-- Depth, shore access, free to dive -->
    <div class="flex gap-1.5 justify-evenly">
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div><!-- Rating and difficulty -->
    <div class="flex items-center justify-evenly">
      <!--v-if-->
      <div class="grow"></div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;
